lldb/tools/lldb-server/Plugins/AMDGPU/ProcessAMDGPU.cpp

```diff
@@ -86,12 +89,64 @@ lldb::addr_t ProcessAMDGPU::GetSharedLibraryInfoAddress() {
   return LLDB_INVALID_ADDRESS;
 }
 
+ThreadAMDGPU *
+ProcessAMDGPU::FindThread(std::function<bool(ThreadAMDGPU &)> pred) {
+  std::lock_guard<std::recursive_mutex> guard(m_threads_mutex);
+  for (ThreadAMDGPU &thread : AMDGPUThreadRange(m_threads)) {
+    if (pred(thread))
+      return &thread;
+  }
+  return nullptr;
+}
+
+static bool ThreadHasValidStopReason(ThreadAMDGPU &thread) {
+  ThreadStopInfo stop_info;
+  std::string stop_description;
+  thread.GetStopReason(stop_info, stop_description);
+
+  return stop_info.reason != lldb::eStopReasonInvalid &&
+         stop_info.reason != lldb::eStopReasonNone;
+}
+
+// Select which thread should be considered the "current thread" in the process.
+lldb::tid_t ProcessAMDGPU::ChooseCurrentThread() {
+  // Helper to find the first thread with a valid stop reason.
+  auto GetFirstThreadWithValidStopReason = [this]() {
+    return FindThread(
+        [](ThreadAMDGPU &thread) { return ThreadHasValidStopReason(thread); });
+  };
+
+  // Return an invalid id when there are no threads.
+  if (m_threads.empty())
+    return LLDB_INVALID_THREAD_ID;
+
+  // If the current thread has a valid stop reason then use that thread.
+  ThreadAMDGPU *current_thread = GetCurrentThreadAMDGPU();
+  if (current_thread && ThreadHasValidStopReason(*current_thread))
```

Comment:
Use the new `NativeThreadProtocol::HasValidStopReason()` method mentioned above.